mesh/src/mutable/Cylindrical2dMesh.cpp

```diff
@@ -160,10 +163,10 @@ void Cylindrical2dMesh::CreateHaloNodes()
     mTopHaloNodes.clear();
     mBottomHaloNodes.clear();
 
-    unsigned num_halo_nodes = (unsigned)(floor(mWidth*2.0));
+    unsigned num_halo_nodes = (unsigned)(floor(mWidth*mHaloScalingFactor));
```

Comment:
Ideally these old c-style casts should be replaced with `static_cast`, e.g.
```cpp
unsigned num_halo_nodes = static_cast<unsigned>(floor(mWidth*mHaloScalingFactor));
```

Reply:
Fixed and updated a few more I caught in the classes.